test/unit/spec/util/jobLogViewerSpec.js

```diff
@@ -48,217 +222,493 @@ define(['util/jobLogViewer', 'common/runtime'], (JobLogViewer, Runtime) => {
-            const viewer = JobLogViewer.make();
-            const jobId = 'testBtnState';
-            const arg = {
-                node: hostNode,
-                jobId: jobId,
-            };
-            await viewer.start(arg);
-            const btns = hostNode.querySelectorAll('div[data-element="header"] button');
-            btns.forEach((btn) => {
-                expect(btn.classList.contains('disabled')).toBeTruthy();
+        describe('initial widget state, history mode on', () => {
+            beforeEach(function () {
+                createLogViewer(this, true);
+            });
+
+            describe('should be determining the state with no job object', () => {
+                beforeEach(async function () {
+                    await this.jobLogViewerInstance.start({
+                        node: this.node,
+                        jobId: 'no state set, history',
+                    });
+                });
+                itHasJobStatusHistory();
+            });
+
+            JobsData.validJobs.forEach((jobState) => {
+                describe(`should create an array in history mode for status ${jobState.status}`, () => {
+                    beforeEach(async function () {
+                        this.jobState = jobState;
+                        await this.jobLogViewerInstance.start({
+                            node: this.node,
+                            jobId: jobState.job_id,
+                            jobState: jobState,
+                        });
+                    });
+                    itHasJobStatusHistory();
+                });
             });
-            viewer.detach();
         });
 
-        it('Should render on job-logs messages immediately on startup', async (done) => {
-            const viewer = JobLogViewer.make();
-            const jobId = 'testJobLogMsgResp';
-            const arg = {
-                node: hostNode,
-                jobId: jobId,
-            };
-            runtimeBus.on('request-job-status', (msg) => {
-                expect(msg).toEqual({ jobId: jobId });
-                runtimeBus.send(
-                    {
-                        jobId: jobId,
-                        jobState: {
-                            status: 'running',
-                        },
-                    },
-                    {
-                        channel: {
-                            jobId: jobId,
-                        },
-                        key: {
-                            type: 'job-status',
-                        },
+        /**
+         * Create a MutationObserver that watches for changes to the job status lines
+         *
+         * @param {object} context - jasmine 'this' context
+         * @param {array} jobMessage - appropriately formatted message and channel data to
+         * @returns {Promise} - resolves when there are job status line changes
+         */
+        function createStatusObserver(context, jobMessage) {
```

Comment:
although there are similar functions in the `TestUtil` file -- `waitForElement`, `waitForElementState`, etc. -- this is specialised enough to need its own function